File: Data Structures & Algorithms/max-water-container/submission-1.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        let res = 0,
            l = 0,
            r = heights.length - 1
        
        while (l < r) {
            let minHeight = Math.min(heights[l], heights[r])
            res = Math.max(res, minHeight * (r - l))
            if (heights[l] < heights[r]) {
                l++
            } else {
                r--
            }
        }

        return res
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit, min_buy = 0, prices[0]

        for sell in prices[1:]:
          max_profit = max(max_profit, sell - min_buy)
          min_buy = min(min_buy, sell)
        
        return max_profit